src/operators.jl

```diff
@@ -201,8 +213,9 @@ function delay{T}(input::Signal{T}, default=value(input); name=auto_name!("delay
 end
 
 function connect_delay(output, input)
-    add_action!(input, output) do output, timestep
-        push!(output, value(input))
+    add_action!(output) do
+        # only push when input is active (avoids it pushing to itself endlessly)
+        input.active && push!(output, value(input))
```

Comment:
You're right. I haven't given much thought to what will break in synchronous case. Will think about it.